src/Service.Tests/GraphQLBuilder/MutationBuilderTests.cs

```diff
@@ -952,38 +1059,50 @@ public static ObjectTypeDefinitionNode GetMutationNode(DocumentNode mutationRoot
             DisplayName = "Mutation name and description validation for singular entity name with singular plural not defined")]
-        [DataRow(GraphQLTestHelpers.PERSON_GQL, "Person", "Person", "People", "Person",
+        [DataRow(GraphQLTestHelpers.PERSON_GQL, new string[] { "Person" }, new string[] { "Person" }, new string[] { "People" }, new string[] { "Person" },
             DisplayName = "Mutation name and description validation for singular entity name with singular plural defined")]
+        [DataRow(GraphQLTestHelpers.PERSON_GQL + GraphQLTestHelpers.BOOK_GQL, new string[] { "Person", "Book" }, new string[] { "Person", "Book" }, new string[] { "People", "Books" }, new string[] { "Person", "Book" },
+            DisplayName = "Mutation name and description validation for multiple entities with singular, plural")]
+        [DataRow(GraphQLTestHelpers.PERSON_GQL + GraphQLTestHelpers.BOOK_GQL, new string[] { "Person", "Book" }, null, null, new string[] { "Person", "Book" },
+            DisplayName = "Mutation name and description validation for multiple entities with singular plural not defined")]
         public void ValidateMutationsAreCreatedWithRightName(
             string gql,
-            string entityName,
-            string singularName,
-            string pluralName,
-            string expectedName
+            string[] entityNames,
+            string[] singularNames,
+            string[] pluralNames,
+            string[] expectedNames
             )
         {
-            DocumentNode root = Utf8GraphQLParser.Parse(gql);
+            Dictionary<string, Entity> entityNameToEntity = new();
+            Dictionary<string, DatabaseType> entityNameToDatabaseType = new();
             Dictionary<string, EntityMetadata> entityPermissionsMap = GraphQLTestHelpers.CreateStubEntityPermissionsMap(
-                    new string[] { entityName },
-                    new EntityActionOperation[] { EntityActionOperation.Create, EntityActionOperation.Update, EntityActionOperation.Delete },
-                    new string[] { "anonymous", "authenticated" });
+                entityNames,
+                new EntityActionOperation[] { EntityActionOperation.Create, EntityActionOperation.Update, EntityActionOperation.Delete },
+                new string[] { "anonymous", "authenticated" });
+            DocumentNode root = Utf8GraphQLParser.Parse(gql);
+
+            for (int i = 0; i < entityNames.Length; i++)
+            {
+                Entity entity = (singularNames is not null)
+                                    ? GraphQLTestHelpers.GenerateEntityWithSingularPlural(singularNames[i], pluralNames[i])
+                                    : GraphQLTestHelpers.GenerateEntityWithSingularPlural(entityNames[i], entityNames[i].Pluralize());
+                entityNameToEntity.TryAdd(entityNames[i], entity);
+                entityNameToDatabaseType.TryAdd(entityNames[i], i % 2 == 0 ? DatabaseType.CosmosDB_NoSQL : DatabaseType.MSSQL);
```

Comment:
for even, why is db type is cosmos and odd it is mssql? any particular reason or random?

Reply:
This was just to create a mix of db types for the test.